src/core/xlio_extra.h

```diff
@@ -641,4 +645,146 @@ static inline struct xlio_api_t *xlio_get_api()
+                  struct timespec *__timeout);
+#endif
+
+ssize_t xlio_recvfrom(int __fd, void *__buf, size_t __nbytes, int __flags, struct sockaddr *__from,
+                      socklen_t *__fromlen);
+
+#if defined HAVE___RECVFROM_CHK
+ssize_t xlio___recvfrom_chk(int __fd, void *__buf, size_t __nbytes, size_t __buflen, int __flags,
+                            struct sockaddr *__from, socklen_t *__fromlen);
+#endif
+
+ssize_t xlio_write(int __fd, __const void *__buf, size_t __nbytes);
+
+ssize_t xlio_writev(int __fd, const struct iovec *iov, int iovcnt);
+
+ssize_t xlio_send(int __fd, __const void *__buf, size_t __nbytes, int __flags);
+
+ssize_t xlio_sendmsg(int __fd, __const struct msghdr *__msg, int __flags);
+
+int xlio_sendmmsg(int __fd, struct mmsghdr *__mmsghdr, unsigned int __vlen, int __flags);
+
+ssize_t xlio_sendto(int __fd, __const void *__buf, size_t __nbytes, int __flags,
+                    const struct sockaddr *__to, socklen_t __tolen);
+
+ssize_t xlio_sendfile(int out_fd, int in_fd, off_t *offset, size_t count);
+
+ssize_t xlio_sendfile64(int out_fd, int in_fd, __off64_t *offset, size_t count);
+
+int xlio_select(int __nfds, fd_set *__readfds, fd_set *__writefds, fd_set *__exceptfds,
+                struct timeval *__timeout);
+
+int xlio_pselect(int __nfds, fd_set *__readfds, fd_set *__writefds, fd_set *__errorfds,
+                 const struct timespec *__timeout, const sigset_t *__sigmask);
+int xlio_poll(struct pollfd *__fds, nfds_t __nfds, int __timeout);
+
+#if defined HAVE___POLL_CHK
+int xlio___poll_chk(struct pollfd *__fds, nfds_t __nfds, int __timeout, size_t __fdslen);
+#endif
+
+int xlio_ppoll(struct pollfd *__fds, nfds_t __nfds, const struct timespec *__timeout,
+               const sigset_t *__sigmask);
+
+#if defined HAVE___PPOLL_CHK
+int xlio___ppoll_chk(struct pollfd *__fds, nfds_t __nfds, const struct timespec *__timeout,
+                     const sigset_t *__sigmask, size_t __fdslen);
+#endif
+
+int xlio_epoll_create(int __size);
+
+int xlio_epoll_create1(int __flags);
+
+int xlio_epoll_ctl(int __epfd, int __op, int __fd, struct epoll_event *__event);
+
+int xlio_epoll_wait(int __epfd, struct epoll_event *__events, int __maxevents, int __timeout);
+
+int xlio_epoll_pwait(int __epfd, struct epoll_event *__events, int __maxevents, int __timeout,
+                     const sigset_t *__sigmask);
+int xlio_socketpair(int __domain, int __type, int __protocol, int __sv[2]);
+
+int xlio_pipe(int __filedes[2]);
+
+int xlio_open(__const char *__file, int __oflag, ...);
+
+int xlio_creat(const char *__pathname, mode_t __mode);
+
+int xlio_dup(int __fd);
+
+int xlio_dup2(int __fd, int __fd2);
+
+pid_t xlio_fork(void);
+
+pid_t xlio_vfork(void);
+
+int xlio_daemon(int __nochdir, int __noclose);
+
+int xlio_sigaction(int signum, const struct sigaction *act, struct sigaction *oldact);
+
+sighandler_t xlio_signal(int signum, sighandler_t handler);
```

Comment:
[Note for future] We should remove these set of prototypes from xlio_extra.h and provide API which solves the same problems: xlio_extra_init() / xlio_extra_terminate()

Reply:
Yes, long term what we really want is a brand new "static library" API with it's own header file that's designed to be efficient, and then separately the POSIX hooks only if you compile as an LD_PRELOADable library.

Reply:
Removed and added xlio_init/xlio_exit with the appropriate comments